### DISCUSSION TITLE
Ultimaker 2 Support

### DISCUSSION THREAD

**Discussion by donovan027** [OrcaSlicer/OrcaSlicer#877] [Q&A]
Hi Everyone. 

I am not sure if this is the right place to ask, so if not, please remove and advise me where to ask.

Would it be possible to add a default machine type / profile for a Ultimaker 2? The profile could be used for a good couple of different machines based on the UM2.

I have tried modifying the Ender 3 profile, but it's not perfect. Would be amazing if the option to add the printer was already there with the correct feeds, speeds, acceleration, etc... 

I have a modified UM2 (SKR Pro 1.2 + Titan Extruder) and a Stock D6. Would be more than happy to test and help where I can.

**Comment by fr3ak2402** [comment]
Hi @stealth027,
theoretically you can create and insert a profile for the Ultimaker 2.

If you could write me the values for the standard Ultimaker 2 here in the chat, I can gladly create a profile for UM2 for you.

**Reply by donovan027** [reply]
That would be amazing! What values are you looking for? Just let me know and I will get hold of them...

**Reply by fr3ak2402** [reply]
I have found a profile for the Ultimat 2 that is intended for the PrusaSlicer (Ultimaten.ini). 
https://github.com/prusa3d/PrusaSlicer/tree/master/resources/profiles
Maybe you can check if the acceleration values etc. fit?

Your Ultimaker is modified, are the values of the standard Ultimaker 2 the same or similar?

**Reply by donovan027** [reply]
That is awesome! Yes, it is modified, but I based the firmware on the stock (Marlin config examples), so the feeds, speeds, and acceleration should be stock standard. Changed extruder steps, LCD, bl touch, etc... but the rest is standard.

**Reply by fr3ak2402** [reply]
Okay, I'm busy with two other printer profiles right now. When these are finished, I will start with the printer profile for the Ultimaker 2.

**Reply by fr3ak2402** [reply]
I think tomorrow I will start your request for an Ultimaker 2.

**Reply by donovan027** [reply]
That would be amazing. Thank you!

**Reply by fr3ak2402** [reply]
Do Marlin GCodes also work on your UltiMaker 2?

**Reply by donovan027** [reply]
Yes. It would seem so. I just sliced (standard profiles) a cube in Cura for the Ultimaker, using the Ultimaker firmware selection and the D6 using the Marlin firmware selection, to compare. The Stock UM2 just handles retraction and material selection on the printer. BUT if you add the printer in PrusaSlicer, then there is the option for material, retraction, etc... and it does work. Hope This helps.

LINK to Gcode files I am referring to: [Ultimaker and D6 Cube.zip](https://github.com/SoftFever/OrcaSlicer/files/11358468/Ultimaker.and.D6.Cube.zip)

**Reply by fr3ak2402** [reply]
Okay, I already implemented some things for the UltiMaker 2 profile last night. I will create a model of the pressure plate and the nozzle later.

When I have finished this, I will import the profile into OrcaSlicer and send you a screenshot.

Can you then go over all the values to see if they are correct or realistic? 

Unfortunately, I don't own an UltiMaker 2, so I don't have any experience with this printer and I can't judge whether the values from the PrusaSlicer profile are correct.

**Reply by donovan027** [reply]
Wonderful! That is amazing, thank you. I will gladly run test prints with the profile. Just let me know what to look for and how you would like me to give you feedback.

**Reply by fr3ak2402** [reply]
Okay, are you using OrcaSlicer on a Windows computer or on a Mac?

Then tomorrow I can tell you how to insert the profile from the Ultimaker into the Orca Slicer for testing.

Before you test, you should take a look at the values for speed and temperatures to see if they fit. Here you can bring in your experience, which values you usually use for these parameters.

**Reply by donovan027** [reply]
I am using it on windows. Sounds like a plan and I look forward to it. Thanks a mill for this, it's really appreciated. 

Will go over the settings that I know are stable and work. Awesome!

**Reply by fr3ak2402** [reply]
I loaded the profile from the Ultimaker 2 into my fork of the OrcaSlicer. The following link will take you to my fork:
https://github.com/fr3ak2402/OrcaSlicer/tree/ultimaker

Just download the code.

Under the ressources folder is a profiles folder. In this folder link a folder UltiMaker and an UltiMaker JSON file, this you need later to test the profile. 

The printer image can also be found under the path ressources /web/image/printer and the file is called UltiMaker 2_cover.png.

In the folder where your OrcaSlicer.exe is located, there is also a ressources. In this folder you will find the same structure. So you copy the UltiMaker folder and the UltiMaker.json into the following path: ressources/profiles/

And the printer image into the following folder ressources/web/image/printer/.

When you have done this, start OrcaSlicer, then you will find the UltiMaker in the overview of all printers. Then you can create it and slice an object like the others.

If you want to make changes in OrcaSlicer, you are welcome to do so. OrcaSlicer will then create a copy of the system profile, which you can then select. 

Important: The profile you have changed will be saved under the following path: 

C:\Users\{Your_Username}\AppData\Roaming\OrcaSlicer\user\default

Don't be surprised, this file saves only changed parameters, these are also only important.

If you need to change filaments or machine profile you can find them also in this path

**Reply by donovan027** [reply]
Hi, I just downloaded the code and I don't see the folder labeled UltiMaker. Did I miss it or am I in the wrong place? I took a screenshot of the folder for you to see.
![image](https://user-images.githubusercontent.com/517499/235354835-ce13edc5-d5b2-464a-a3e7-91c88085c0c9.png)

**Reply by fr3ak2402** [reply]
Oh my mistake. I made some changes to the main branch earlier, I moved this to the ultimaker branch.

https://github.com/fr3ak2402/OrcaSlicer/tree/ultimaker

important: if you go into my fork, the branch must be changed to ultimaker. then you can download the code and then also have the Ultimaker profile folders

<img width="933" alt="Bildschirmfoto 2023-04-30 um 15 35 56" src="https://user-images.githubusercontent.com/96239814/235355812-8f1ce6db-07d0-4e81-92ae-b2e8601012c5.png">

**Comment by donovan027** [comment]
Ok. Got it and installed it. Will start playing with the profile and see what I can get. Thanks for this. Appreciate it.

**Reply by fr3ak2402** [reply]
Okay, just tell me the changes you made to the profile.

Have you been able to print a test and what did it look like?

**Reply by donovan027** [reply]
Hi. I printed a couple of benchy's to get it tuned. Overhangs still need a little work, but that's a UM2 thing. The stock cooling lacks a lot to be desired. Reduced the speed drastically. The top speed is 300mm/s. It will never happen, lol. Worked on a UM3 and UM5 before and even with those, we ran a lot slower. For a decent standard print, it's more like 40mm/s. Haven't messed with the Fine or Draft yet. going to do the Orca Cube v2 now and see how it turns out. Can I upload you a excel sheet or txt document of the settings I am using? Will also upload a pic of the benchy and the cube.

**Reply by fr3ak2402** [reply]
Hi, thanks for revising the profile. Yes, just upload an Excel spreadsheet, that's the easiest for me.

It is important for me that you also write me the parameter name of the profile, that makes the whole thing easier for me.

For example, for the layer height:

<img width="554" alt="Bildschirmfoto 2023-05-01 um 12 18 35" src="https://user-images.githubusercontent.com/96239814/235440085-2b8262e8-20f9-4ad8-bc85-fc9a84a64f7e.png">

If I understood correctly, you still have a UM3 and UM5. Should I also create a profile for these two UltiMaker models later?

**Reply by donovan027** [reply]
Will do. Will get everything for the 'Standard' profile so long into a spreadsheet for you. No, I personally don't have a UM3 or UM5. I used to work at a company that had them and worked on and with them.

**Reply by fr3ak2402** [reply]
Okay. Will you do the values for the Fine & Draft profile later?

**Reply by donovan027** [reply]
I definitely will do the fine and draft and will keep on tuning them until it's the best I can get it. I have to do some work today (need to pay the bills, lol), and then it's back to work tomorrow, but I will make sure that I get them done though. Will do a little every night between the norm. But, I will upload the normal profile today. Could obviously still be fine-tuned, but it would be a solid starting point with more than acceptable prints.

**Reply by fr3ak2402** [reply]
Hi @stealth027,
Have you perhaps already made some progress in testing?

**Reply by donovan027** [reply]
Hi, no, sorry I have not. I had to leave urgently to a remote site in Limpopo (SA) for work. Will be heading home (Pretoria, SA) on Friday and will see how far I can get with all of the profiles over the weekend. Sorry about that. It's been insane.

**Reply by fr3ak2402** [reply]
Hi, oh okay. Could you already do some testing and write it to me in an excel spreadsheet?

**Reply by donovan027** [reply]
Sorry, been working nonstop. But, here is what I am using so far and getting decent results. Not perfect, but more than acceptable. Bussy with a litho now on 0.12. Looking good. Will upload the fine profile tomorrow night when I get back from work.
P.S. Serious overhangs and bridging are still an issue, but it's the stock cooling.
[UM2 Print Settings.xlsx](https://github.com/SoftFever/OrcaSlicer/files/11415754/UM2.Print.Settings.xlsx)

**Reply by fr3ak2402** [reply]
I can also wait until you have completed the whole test.

Or I can adjust your changes in the profile and publish the part for the next release of the OrcaSlicer.

The next changes from you I can then incorporate into the next version.

**Comment by donovan027** [comment]
Hi, I think that would be best. That way, that type of machine is at least in and people can start adding and playing. I will carry on with the profiles and get them done though. At the moment, sitting in thunderstorms and the internet in down. But, will get them done ASAP.

**Reply by fr3ak2402** [reply]
Okay, let's do it this way. Thanks for all the work for testing. I will also have you included in changenotes of the new version of OrcaSlicer. Since your work on the Ultimaker Profile should also be appreciated.

**Comment by donovan027** [comment]
Hi, quick question. I wanted to start on the profiles again tonight and am getting an Unhandled unknown exception; terminating the application. What could be the cause of this?

I also downloaded the actual release and its doing the same thing. 

![Screenshot 2023-05-10 191152](https://github.com/SoftFever/OrcaSlicer/assets/517499/82dd75b8-4449-46fb-ad19-6b32c9915d95)

![Screenshot 2023-05-10 190946](https://github.com/SoftFever/OrcaSlicer/assets/517499/c5f231ba-3f73-41f9-9b85-c89eb010bb98)

**Reply by fr3ak2402** [reply]
Did you perhaps delete something from this folder?
Or changed a system setting of Windows?

Otherwise just rename the following folder from OrcaSlicer to OrcaSlider2

You can find the folder to rename under the following path:

C:\Users\donovlAppData\Roaming\

Then restart OrcaSlicer and the setup window should appear again.
Once you have everything set up again, simply close OrcaSlicer and copy your custom configurations back into the following folder of the new OrcaSlicer folder.

These folders are located under the following path:
C:\Users\donovlAppData\Roaming\OrcaSlicer\user\default\

**Reply by fr3ak2402** [reply]
@stealth027 

Hi and is your OrcaSlicer working again?

**Reply by donovan027** [reply]
Hi, yes it is, thanks. Downloaded a release and put the files from your branch in it and it's working. I am in Swartklip (Limpopo, South Africa) at the moment for work. Will only be back home on Saturday the 27th though. Will start with the profiles again as soon as I am back.

**Comment by DylanReynolds3Shape** [comment]
Hello,

Is the Ultimaker listed in the supported machines the Ultimake 2+ connect?

Thanks,

Dylan
